src/Orleans.Runtime/GrainDirectory/CachedGrainLocator.cs

```diff
@@ -105,7 +104,7 @@ public async Task<ActivationAddress> Register(ActivationAddress address)
             }
 
             // Cache update
-            this.cache.AddOrUpdate(activationAddress, 0);
+            this.cache.AddOrUpdate(activationAddress, (int) result.MembershipVersion.Value);
```

Comment:
Should we change `cache` to hold `MembershipVersion` instances?

Reply:
Yes, but in another PR, I don't want to touch the logic from the in-cluster directory just now